rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/util/FormKeyTextExtractionUtilTest.java

```diff
@@ -0,0 +1,73 @@
+package io.ona.rdt.robolectric.util;
+
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.widgets.RepeatingGroupFactory;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.AllConstants;
+import org.smartregister.util.JsonFormUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import io.ona.rdt.robolectric.RobolectricTest;
+import io.ona.rdt.util.CovidConstants;
+import io.ona.rdt.util.FormKeyTextExtractionUtil;
+
+public class FormKeyTextExtractionUtilTest extends RobolectricTest {
+
+    private static final Set<String> FORMS_TO_EXTRACT_TEXT_FROM = new HashSet<>(Arrays.asList(
+            CovidConstants.Form.PATIENT_DIAGNOSTICS_FORM, CovidConstants.Form.SAMPLE_COLLECTION_FORM,
+            CovidConstants.Form.COVID_RDT_TEST_FORM, CovidConstants.Form.XRAY_FORM, CovidConstants.Form.WBC_FORM));
+
+    @Test
+    public void testGetFormWidgetKeyToTextMapShouldPopulateNonEmptyMap() throws Exception {
+
+        Map<String, String> formWidgetKeyToTextMap = new HashMap<>();
+        for (String formName : FORMS_TO_EXTRACT_TEXT_FROM) {
```

Comment:
Any reason to do it this way instead of using the util logic to perform the extraction?

Reply:
I have called `FormKeyTextExtractionUtil.getFormWidgetKeyToTextMap()` for actual data and use this entire loop for expected data.

Reply:
This is duplicating the logic already in the `FormKeyTextExtractionUtil` which is not recommended. It should be enough to call `getFormWidgetKeyToTextMap()` and probably inject a mock `FORMS_TO_EXTRACT_TEXT_FROM` set. Then verify that some fields from those forms are extracted into the map.

Reply:
We should especially verify that the special widgets e.g. repeating group, checkbox, radio buttons are correctly processed.

Reply:
I have used a predefined map.

Reply:
remove the map and just check a few keys to verify that the values match. Include a key from the option, checkbox, repeating group and regular text fields.

Reply:
I have checked a few keys and values but I think we still need a map to hold keys and values.